src/test/java/de/tum/in/www1/artemis/service/notifications/SingleUserNotificationServiceTest.java

```diff
@@ -1,109 +1,72 @@
 
-    @Mock
-    private static SingleUserNotificationRepository singleUserNotificationRepository;
+    @Autowired
+    private SingleUserNotificationService singleUserNotificationService;
 
-    @Mock
-    private static SimpMessageSendingOperations messagingTemplate;
+    @Autowired
+    private NotificationRepository notificationRepository;
 
-    @Mock
-    private static MailService mailService;
+    @Autowired
+    private NotificationSettingRepository notificationSettingRepository;
 
-    @Mock
-    private static UserRepository userRepository;
-
-    @Mock
-    private static NotificationSettingsService notificationSettingsService;
-
-    private static Post post;
-
-    private static Lecture lecture;
+    private Notification capturedNotification;
 
-    private static Course course;
+    private User user;
 
-    private static final Long COURSE_ID = 27L;
+    private FileUploadExercise fileUploadExercise;
 
-    private static PlagiarismComparison plagiarismComparison;
+    private Post post;
 
-    private static PlagiarismSubmission plagiarismSubmission;
+    private Course course;
 
-    private static PlagiarismResult plagiarismResult;
+    private PlagiarismComparison plagiarismComparison;
 
     /**
-     * Sets up all needed mocks and their wanted behavior once for all test cases.
-     * These are the common mocks and structures which behavior is fixed and will not change
+     * Sets up all needed mocks and their wanted behavior
      */
-    @BeforeAll
-    public static void setUp() {
-        mailService = mock(MailService.class);
-        doNothing().when(mailService).sendNotificationEmailForMultipleUsers(any(), any(), any());
-
-        course = new Course();
-        course.setId(COURSE_ID);
-
-        user = new User();
-
-        userRepository = mock(UserRepository.class);
-        when(userRepository.getUser()).thenReturn(user);
-
-        notificationCaptor = ArgumentCaptor.forClass(Notification.class);
-
-        messagingTemplate = mock(SimpMessageSendingOperations.class);
-
-        notificationSettingsService = mock(NotificationSettingsService.class);
+    @BeforeEach
+    public void setUp() {
+        SecurityUtils.setAuthorizationObject();
 
-        singleUserNotificationRepository = mock(SingleUserNotificationRepository.class);
+        course = database.createCourse();
 
-        singleUserNotificationService = spy(
-                new SingleUserNotificationService(singleUserNotificationRepository, userRepository, messagingTemplate, mailService, notificationSettingsService));
+        List<User> users = database.addUsers(1, 0, 0, 0);
+        user = users.get(0);
 
-        exercise = new TextExercise();
+        Exercise exercise = new TextExercise();
         exercise.setCourse(course);
 
         fileUploadExercise = new FileUploadExercise();
         fileUploadExercise.setCourse(course);
 
-        lecture = new Lecture();
+        Lecture lecture = new Lecture();
```

Comment:
`Modelfactory.generateLecture()` would also be considerable here

Reply:
have a look at my first response